magenta/music/musicxml_parser.py

```diff
@@ -465,6 +480,58 @@ def _parse_forward(self, xml_forward):
                * self.state.seconds_per_quarter)
     self.state.time_position += seconds
 
+  def check_time_signature(self):
+    """Correct the time signature for incomplete measures.
+
+    If the measure is incomplete or a pickup, insert an appropriate
+    time signature into this Measure.
+    """
+    # Compute the fractional time signature (duration / divisions)
+    fractional_time_signature = Fraction(self.duration, self.state.divisions)
```

Comment:
it looks like this value is used only in the `else` block. might make sense to move it there.